app/screens/login/form.tsx

```diff
@@ -117,22 +117,24 @@ const LoginForm = ({config, extra, keyboardAwareRef, serverDisplayName, launchEr
         resetToHome({extra, launchError: hasError, launchType, serverUrl});
     };
 
-    const checkLoginResponse = (data: LoginActionResponse) => {
-        let errorId = '';
-        const loginError = data.error;
-        if (isServerError(loginError) && loginError.server_error_id) {
-            errorId = loginError.server_error_id;
+    const isMFAError = (loginError: unknown): boolean => {
+        const serverError = getServerError(loginError);
+        if (serverError) {
+            return MFA_EXPECTED_ERRORS.includes(serverError);
         }
+        return false;
+    };
 
-        if (data.failed && MFA_EXPECTED_ERRORS.includes(errorId)) {
+    const checkLoginResponse = (data: LoginActionResponse) => {
+        if (data.failed && isMFAError(data.error)) {
```

Comment:
very (10x) nit:

```
        const {failed, error} = data
        if (failed && isMFAError(error)) {
            goToMfa();
            setIsLoading(false);
            return false;
        }

        if (failed && error) {
            setIsLoading(false);
            setError(getLoginErrorMessage(error));
            return false;
        }
```

Reply:
there you go, couldn't name it `error` as that is already used by `useState`